h2integrate/core/utilities.py

```diff
@@ -290,3 +293,163 @@ def dict_to_yaml_formatting(orig_dict):
+        names = re.sub(" ", "", f.readline()).strip().split(",")
+        name_map = {name: name for name in names if name in design_vars}
+
+    invalid_desvars = [name for name in names if name not in name_map]
+
+    if check_only:
+        if len(invalid_desvars) == 0:
+            return True
+        else:
+            return False
+
+    if len(invalid_desvars) == 0:
+        return csv_fpath
+
+    file_txt_to_remove = []
+    remove_index = False
+    if len(invalid_desvars) > 0:
+        for invalid_var in invalid_desvars:
+            if invalid_var != "":
+                contains_dvar = [d for d in design_vars if d in invalid_var]
+                if len(contains_dvar) == 1:
+                    txt_to_remove = [
+                        rm_txt for rm_txt in invalid_var.split(contains_dvar[0]) if rm_txt != ""
+                    ]
+                    file_txt_to_remove.extend(txt_to_remove)
+
+                if len(contains_dvar) > 1:
+                    # duplicate definitions of the design variable
+                    msg = (
+                        f"{invalid_var} is does not match a unique design variable. The design "
+                        f"variables defined in the driver_config file are {design_vars}."
+                        f" Please check the csv file {csv_fpath} to only have one column per "
+                        "design variable included in the driver config file."
+                    )
+                    raise ValueError(msg)
+                if len(contains_dvar) == 0:
+                    msg = (
+                        f"{invalid_var} is an invalid design variable. The design "
+                        f"variables defined in the driver_config file are {design_vars}."
+                        f" Please check the csv file {csv_fpath} to only have the design "
+                        "variables included in the driver config file."
+                    )
+                    raise ValueError(msg)
+
+            else:
+                remove_index = True
+                with Path(csv_fpath).open() as f:
+                    reader = csv.DictReader(f)
+                    index_col = [i for i, n in enumerate(reader.fieldnames) if n == ""]
+                    # for row in reader:
+                    #     for name in reader.fieldnames:
+                    #         if name=="":
+                    #             npfs = np.fromstring(re.sub(r'[\[\]]', '', row[name]), sep=' ')
+
+        original_file = Path(csv_fpath).open()
+        lines = original_file.readlines()
+        original_file.close()
+        for f_remove in file_txt_to_remove:
+            lines = [line.replace(f_remove, "") for line in lines]
+        if remove_index:
+            # remove the columns that are index columns
+            lines = [
+                ",".join(lp for li, lp in enumerate(line.split(",")) if li not in index_col)
+                for line in lines
+            ]
```

Comment:
Could you please add a bit more in-line comments on why you're doing what you're doing here, what you're checking for, what the fixes are, etc? Just so when folks look at the logic it'll be more clear what's going on.